lms/djangoapps/verify_student/models.py

```diff
@@ -1262,6 +1265,51 @@ def check_user_has_completed_checkpoint(cls, user_id, course_key, related_assess
             status__in=[cls.SUBMITTED_STATUS, cls.APPROVED_STATUS]
         ).exists()
 
+    @classmethod
+    def get_all_checkpoints(cls, user_id, course_key):
+        """Return dict of all the checkpoints with their status.
+        Args:
+            user_id(int): Id of user.
+            course_key(unicode): Unicode of course key
+
+        Returns:
+            dict: {checkpoint:status}
+        """
+        cache_key = VerificationStatus.cache_key_name(user_id, unicode(course_key))
+        all_checks_points = cls.objects.filter(
+            user_id=user_id, checkpoint__course_id=course_key
+        )
+        check_points = {}
+        for check in all_checks_points:
+            check_points[check.checkpoint.checkpoint_location] = check.status
+
+        cache.set(cache_key, check_points)
```

Comment:
It's a little bit strange to be setting the cache here, since we're not retrieving any values from the cache.  I think it would be clearer to move the `cache.set()` calls to `get_group_for_user()` where they are accessed.